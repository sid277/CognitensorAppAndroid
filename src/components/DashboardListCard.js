import React from 'react';
import { Text, StyleSheet, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { theme } from '../theme';

const colors = [
  '#FF8DD7',
  '#0BCE96',
  '#FADB6D',
  '#8BBFFF',
  '#FFA67D',
  '#FF4E4E',
];

const DashboardListCard = ({ item, index }) => {
  const color = colors[index % colors.length];
  return (
    <View style={styles.item}>
      <View style={[styles.itemLeftIcon, { backgroundColor: color }]}>
        <MaterialIcons name="bar-chart" size={24} color="#FFFFFF" margin={4} />
      </View>
      <Text style={styles.itemFont}>
        {item.dashboardTitle || item.dashboardName}
      </Text>
      <View style={styles.itemRightIcon}>
        <MaterialIcons name="chevron-right" size={24} color="#4A4A4A" />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  item: {
    flexDirection: 'row',
    backgroundColor: '#FEFFFF',
    padding: theme.spacing.base,
    marginVertical: theme.spacing.tiny,
    marginHorizontal: theme.spacing.small,
    borderRadius: 8,
    borderWidth: 0.5,
    borderColor: '#F5EAF1',
    // Shadow for iOS
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    // Shadow for android
    elevation: 1,
  },
  itemFont: {
    ...theme.typography.headline,
    marginLeft: theme.spacing.tiny,
  },
  itemLeftIcon: {
    justifyContent: 'flex-start',
    borderRadius: 50,
  },
  itemRightIcon: {
    justifyContent: 'flex-end',
    marginLeft: 'auto',
  },
});

export default DashboardListCard;
