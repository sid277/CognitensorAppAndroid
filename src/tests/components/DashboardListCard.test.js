import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import DashboardListCard from '../../components/DashboardListCard';

test('should test DashboardListCard component', () => {
  const item = {
    dashboardTitle : "Dashboard_test1"
  }
  const wrapper = shallow(<DashboardListCard item = {item}/>);
  expect(wrapper).toMatchSnapshot();
  });

// it('renders DashboardListCard component', () => {
//   const tree = renderer.create(<DashboardListCard/>).toJSON();
//   expect(tree).toMatchSnapshot();
// });
