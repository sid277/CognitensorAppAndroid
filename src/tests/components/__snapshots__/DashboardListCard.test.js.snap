// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test DashboardListCard component 1`] = `ShallowWrapper {}`;
